src/Monitors/RadioMonitor.cpp

```diff
@@ -177,32 +157,68 @@ void RadioMonitor::receive()
         Serial.print(radio->getFrequencyError());
         Serial.println(F(" Hz"));
 
-    } else if (code == RADIOLIB_ERR_RX_TIMEOUT) {
-        // timeout occurred while waiting for a packet
-        Serial.println(F("timeout!"));
-
-    } else if (code == RADIOLIB_ERR_CRC_MISMATCH) {
-        // packet was received, but is malformed
-        Serial.println(F("CRC error!"));
-
+        return true;
     } else {
-        // some other error occurred
-        Serial.print(F("failed, code "));
-        Serial.println(code);
+        if (code == constants::radio::err_rx_timeout) {
+            // timeout occurred while waiting for a packet
+            Serial.println(F("timeout!"));
+
+        } else if (code == constants::radio::err_crc_mismatch) {
+            // packet was received, but is malformed
+            Serial.println(F("CRC error!"));
+
+        } else {
+            // some other error occurred
+            Serial.print(F("failed, code "));
+            Serial.println(code);
+        }
+        return false;
     }
 }
 
 void RadioMonitor::execute()
 {
-    // just form report here to make life easy
-    switch (sfr::radio::receive_mode)
-
-        if (millis() - sfr::radio::last_receive >= sfr::radio::receive_period) {
-            // receive every minutes- add bit to end of downlink report indicating we are in receive mode
-            receive();
+    // implements the state machine described in: https://github.com/Alpha-CubeSat/oop-chipsat-code/wiki
+    switch (sfr::radio::mode) {
+    case radio_mode_type::init:
+        Serial.println(F("Init State"));
+        init();
+        if (sfr::radio::init_mode == sensor_init_mode_type::complete) {
+            sfr::radio::mode = radio_mode_type::waiting;
+            sfr::radio::listen_period_start = millis();
+            sfr::radio::downlink_period_start = millis();
         }
-
-    // transmit();
+        break;
+    case radio_mode_type::waiting:
+        Serial.println(F("Waiting State"));
+        if (millis() - sfr::radio::listen_period_start >= sfr::radio::listen_period) {
+            sfr::radio::mode = radio_mode_type::listen;
+            sfr::radio::command_wait_start = millis();
+        } else if (millis() - sfr::radio::downlink_period_start >= sfr::radio::downlink_period) {
+            sfr::radio::mode = radio_mode_type::downlink;
+        }
+        break;
+    case radio_mode_type::downlink:
+        Serial.println(F("Downlink State"));
+        // placeholder for reports
+        byte byteArr_t[] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+        bool transmit_success = transmit(byteArr_t, 8);
+        if (transmit_success) {
+            sfr::radio::mode = radio_mode_type::waiting;
```

Comment:
What is we never transmit successfully? The way I think this is currently written, we are never able to process uplinks since we never reenter waiting mode.